packages/core/database/src/migrations/internal-migrations/index.ts

```diff
@@ -0,0 +1,12 @@
+import type { Migration } from '../common';
+
+/**
+ * List of all the internal migrations. The array order will be the order in which they are executed.
+ *
+ * {
+ *   name: 'some-name',
+ *   async up(knex: Knex, db: Database) {},
+ *   async down(knex: Knex, db: Database) {},
+ * },
+ */
+export const internalMigrations: Migration[] = [];
```

Comment:
I was thinking how we can organize this file (/folder) to know which packages add which migrations.

My first thought was to create a subfolder for every package, and there add every migration file.
For that we would probably need to add a `createdAt`: `Date` in the `Migration` interface, to be able to sort the migrations.

wdyt?

Reply:
On a similar line, what if we allow packages of the monorepo to create an internal-migrations folder, and we read from all of them on startup? That would allow other packages to own its own internal migrations, use their own types , logic ...

Reply:
- Ordering is managed by the array itself. I think it will be much simpler to control the order this way than having to play around with dates. if we have conflcits between PRs we will see it in the array order when we would miss it with a sortBy date. 

- We can add this but this isn't internal migrations but plugin/external migrations. we can expose a programmatic API to register migrations (that a user could also use in that case) but I wonder if use the migration hooks or adding new one shouldn't be prefered. To be discussed in another PR at least.

Reply:
makes sense, this already unblocks having internal migration, so let's see from here how it goes 🚀